api/src/identity-access-management/domain/emails/create-reset-password-demand.email.js

```diff
@@ -0,0 +1,43 @@
+import { LOCALE } from '../../../shared/domain/constants.js';
+import { urlBuilder } from '../../../shared/infrastructure/utils/url-builder.js';
+import { EmailFactory } from '../../../shared/mail/domain/models/EmailFactory.js';
+import { mailer } from '../../../shared/mail/infrastructure/services/mailer.js';
+
+/**
+ * Creates a reset password demand email.
+ *
+ * @param {Object} params - The parameters for the email.
+ * @param {string} params.email - The recipient's email address.
+ * @param {string} params.temporaryKey - The temporary key for resetting the password.
+ * @param {string} params.locale - The locale for the email.
+ * @returns {Email} - The email instance.
+ */
+export function createResetPasswordDemandEmail({ email, temporaryKey, locale = LOCALE.FRENCH_FRANCE }) {
+  const factory = new EmailFactory({ app: 'pix-app', locale });
+
+  const { i18n, defaultVariables } = factory;
+
+  const pixAppUrl = urlBuilder.getPixAppBaseUrl(locale);
+
+  return factory.buildEmail({
+    template: mailer.passwordResetTemplateId,
+    subject: i18n.__('reset-password-demand-email.subject'),
+    to: email,
+    variables: {
+      locale,
+      homeName: defaultVariables.homeName,
+      homeUrl: defaultVariables.homeUrl,
+      helpdeskUrl: defaultVariables.helpdeskUrl,
+      resetUrl: `${pixAppUrl}/changer-mot-de-passe/${temporaryKey}/?lang=${locale}`,
+      clickOnTheButton: i18n.__('reset-password-demand-email.clickOnTheButton'),
```

Comment:
Il manque le `.params` dans les appels des trads 😉 

```suggestion
      clickOnTheButton: i18n.__('reset-password-demand-email.params.clickOnTheButton'),